repose-aggregator/core/repose-core/src/test/scala/org/openrepose/nodeservice/response/ResponseHeaderServiceImplTest.scala

```diff
@@ -0,0 +1,80 @@
+
+import java.util.Optional
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+
+import org.junit.runner.RunWith
+import org.mockito.Matchers.anyString
+import org.mockito.Mockito.{never, verify, when}
+import org.openrepose.commons.utils.http.CommonHttpHeader
+import org.openrepose.nodeservice.containerconfiguration.ContainerConfigurationService
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfterEach, FunSpec, Matchers}
+
+import scala.util.Random
+
+@RunWith(classOf[JUnitRunner])
+class ResponseHeaderServiceImplTest extends FunSpec with BeforeAndAfterEach with Matchers with MockitoSugar {
+
+  private val version = "9.8.7.6"
+  private var request: HttpServletRequest = _
+  private var response: HttpServletResponse = _
+  private var containerConfigurationService: ContainerConfigurationService = _
+  private var responseHeaderServiceImpl: ResponseHeaderServiceImpl = _
+
+  override def beforeEach(): Unit = {
+    request = mock[HttpServletRequest]
+    response = mock[HttpServletResponse]
+    containerConfigurationService = mock[ContainerConfigurationService]
+    responseHeaderServiceImpl = new ResponseHeaderServiceImpl(containerConfigurationService, version)
+  }
+
+  describe("The Response Header Service Implementation") {
+    val local = Math.abs(Random.nextInt)
+    val shouldOrNot: Boolean => String = { boolean => if (boolean) "should" else "should not" }
+    val prefixToString: String => String = { string => if (string != null && string.length > 0) s" $string" else "" }
+    val versionString: Boolean => String = { boolean => if (boolean) s" (Repose/$version)" else "" }
+    Seq("1.0", "1.1") foreach { protocol =>
+      Seq("prefix", "", null) foreach { prefix =>
+        Seq(true, false) foreach { includeVersion =>
+          it(s"${shouldOrNot((prefix != null && prefix.length > 0) || includeVersion)} set the Via header with protocol '$protocol', prefix '$prefix', and includeVersion '$includeVersion'") {
+            when(request.getLocalPort).thenReturn(local)
+            when(request.getProtocol).thenReturn(s"HTTP/$protocol")
+            when(containerConfigurationService.getResponseVia).thenReturn(Optional.ofNullable(prefix))
+            when(containerConfigurationService.includeViaReposeVersion()).thenReturn(includeVersion)
+
+            responseHeaderServiceImpl.setVia(request, response)
+
+            if ((prefix == null || prefix.length == 0) && includeVersion) {
+              verify(response).setHeader(CommonHttpHeader.VIA, s"$protocol Repose${versionString(includeVersion)}")
+            } else if ((prefix != null && prefix.length > 0) || includeVersion) {
+              verify(response).setHeader(CommonHttpHeader.VIA, s"$protocol${prefixToString(prefix)}${versionString(includeVersion)}")
+            } else {
+              verify(response, never).setHeader(anyString, anyString)
+            }
```

Comment:
**DAMP** (Descriptive And Meaningful Phrases) and **DRY**:
http://stackoverflow.com/a/11837973/3448480